x-pack/plugins/enterprise_search/server/routes/app_search/app_search_request_handler.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import fetch from 'node-fetch';
+import querystring from 'querystring';
+import { Logger } from 'src/core/server';
+import { ConfigType } from '../../index';
+
+interface IAppSearchRequestParams {
+  config: ConfigType;
+  log: Logger;
+  path: string;
+  hasValidData: (body) => boolean;
+}
+
+/**
+ * This function creates a handler, in a standard way, for handling App Search
+ * API requests.
+ *
+ * This handler assumes that it will essentially just proxy the App Search API
+ * request, so the request body and request parameters, and body are simply
+ * passed through.
+ */
+export function createAppSearchRequestHandler({
+  config,
+  log,
+  path,
+  hasValidData,
+}: IAppSearchRequestParams) {
+  return async (_context, request, response) => {
```

Comment:
I only just realized that I'm not super clear on this - but where are these args coming from? Is there some Kibana logic somewhere that's injecting these vars?

Reply:
Typing fixes that (I think) should work, feel free to verify

```js
import {
  RequestHandlerContext,
  KibanaRequest,
  KibanaResponseFactory,
} from 'src/core/server';


...

  return async (_context: RequestHandlerContext, request: KibanaRequest, response: KibanaResponseFactory) => {
```

Based on examples I'm seeing in Kibana the request _may_ have to be `request: KibanaRequest<unknown, unknown, unknown>` but feel free to try without first